subsys/console/getchar.c

```diff
@@ -13,14 +13,36 @@ static struct tty_serial console_serial;
 static u8_t console_rxbuf[CONFIG_CONSOLE_GETCHAR_BUFSIZE];
 static u8_t console_txbuf[CONFIG_CONSOLE_PUTCHAR_BUFSIZE];
 
+ssize_t console_write(void *dummy, const void *buf, size_t size)
+{
+	ARG_UNUSED(dummy);
+
+	return tty_write(&console_serial, buf, size);
+}
+
+ssize_t console_read(void *dummy, void *buf, size_t size)
+{
+	ARG_UNUSED(dummy);
+
+	return tty_read(&console_serial, buf, size);
+}
+
 int console_putchar(char c)
 {
-	return tty_putchar(&console_serial, (u8_t)c);
+	return tty_write(&console_serial, &c, 1);
 }
 
-u8_t console_getchar(void)
+int console_getchar(void)
 {
-	return tty_getchar(&console_serial);
+	u8_t c;
+	int res;
+
+	res = tty_read(&console_serial, &c, 1);
+	if (res < 0) {
+		return -1;
```

Comment:
why `-1` and not `ret`?

Reply:
> why -1 and not res?

Kinda wanted it to mimic behavior of ANSI C, which guarantees that return value has type of set {EOF, 0-255} (not an arbitrary negative value). But ok, let's just propagate `res` for now.